api/novelist.py

```diff
@@ -461,54 +464,83 @@ def get_isbns_from_query(self, library):
         i2 = aliased(Identifier)
 
         isbnQuery = select(
-            [i1.identifier, i1.type, i2.identifier]
+            [i1.identifier, i1.type, i2.identifier,
+            Edition.title, Edition.medium, Edition.sort_author, Edition.author,
+            Contribution.role, Contributor.sort_name],
         ).select_from(
             join(LicensePool, i1, i1.id==LicensePool.identifier_id)
             .join(Equivalency, i1.id==Equivalency.input_id, LEFT_OUTER_JOIN)
             .join(i2, Equivalency.output_id==i2.id, LEFT_OUTER_JOIN)
+            .join(
+                Edition,
+                or_(Edition.primary_identifier_id==i1.id, Edition.primary_identifier_id==i2.id)
+            )
+            .join(Contribution, Edition.id==Contribution.edition_id)
+            .join(Contributor, Contribution.contributor_id==Contributor.id)
         ).where(
             and_(
                 LicensePool.collection_id.in_(collectionList),
-                or_(i1.type=="ISBN", i2.type=="ISBN")
+                or_(i1.type=="ISBN", i2.type=="ISBN"),
+                or_(Contribution.role=="Primary Author", Contribution.role=="Narrator"),
```

Comment:
I added `"Primary Author"` check here because there were multiple rows of the same ISBN but the only difference was this role value which could've been "Author", "Primary Author", "Unknown", "Illustrator", etc.

Reply:
You also need to pick up "Author", for the case where a book has multiple authors or no primary author is specified. So instead of this `or` clause, use a clause defined in the following way:

```
roles = list(Contributor.AUTHOR_ROLES)
roles.append(Contributor.NARRATOR_ROLE)
clause = Contribution.role.in_(roles)
```